instrumentation/netty/netty-3.8/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/netty/v3_8/ChannelTraceContext.java

```diff
@@ -74,4 +75,16 @@ public boolean equals(Object obj) {
   public int hashCode() {
     return Objects.hash(connectionContext, clientParentContext, context);
   }
+
+  @Override
+  public String toString() {
```

Comment:
Is that `toString()` needed?

Reply:
There's a check that toString is added if equals/hashcode are. I can disable it if it seems to much but figure having a toString is generally better than not.

Reply:
Hmm, it's not really a big deal or anything, but those methods will probably be never called - they're just dead code. If we need a `toString()` method we usually add it.

Reply:
I was going to say +1 for disabling this rule, then went to read why the rule exists, and didn't realize `Object.toString()` was affected by overriding `hashCode()` 😱, I guess I thought it was always based on `System.identityHashCode()`, so now at least the rule makes sense to me. I think I'm probably still _slightly_ +1 for disabling the rule for the same reasons as @mateuszrzeszutek mentioned, but also good with keeping it if you think it's useful.

Reply:
IntelliJ calls toString when debugging which is the only reason really to have toString on types that won't be called by code. Not that debugging works well for us :) But it's why I slightly lean towrads keeping it